ci/jenkins/test-vm.sh

```diff
@@ -139,26 +142,72 @@ function apply_antrea {
-    for i in `seq 10`; do
-        WINDOWS_VM_IP=$(govc vm.ip -k=true -wait=2m ${WIN_HOSTNAME})
-        if [[ WINDOWS_VM_IP == "" ]]; then
-            echo "Failed to retrieve IP for Windows VM ${WIN_HOSTNAME}, retry ${i}"
-            continue
-        fi
+function export_govc_env_var {
+    # This should be coming from jenkins configuration
+    export GOVC_URL=$GOVC_URL
+    export GOVC_USERNAME=$GOVC_USERNAME
+    export GOVC_PASSWORD=$GOVC_PASSWORD
+    export GOVC_INSECURE=1
+    export GOVC_DATACENTER=$GOVC_DATACENTER
+    export GOVC_DATASTORE=$GOVC_DATASTORE
+}
+
+function clean_up_one_ns {
+    ns=$1
+    kubectl get pod -n "${ns}" --no-headers=true | awk '{print $1}' | while read pod_name; do
+        kubectl delete pod "${pod_name}" -n "${ns}" --force --grace-period 0
     done
+    kubectl delete ns "${ns}" --ignore-not-found=true || true
+}
 
+function clean_service_account_config {
     kubectl delete sa $SERVICE_ACCOUNT -n $TEST_NAMESPACE --ignore-not-found=true
     clean_up_one_ns $TEST_NAMESPACE
-    echo "Deleting stale antrea-agent files from ${LIN_HOSTNAME} and ${WIN_HOSTNAME}"
-    ssh -o StrictHostKeyChecking=no -i "${WORKDIR}/jenkins_id_rsa" -n ubuntu@${UBUNTU_VM_IP} "rm -rf /tmp/antrea-ci"
-    ssh -o StrictHostKeyChecking=no -i "${WORKDIR}/jenkins_id_rsa" Administrator@${WINDOWS_VM_IP} "rm -rf /tmp/antrea-ci"
+}
+
+function clean_vm_agent_on_linux {
+    for host_name in "${!LINUX_HOSTS_TO_IP[@]}"; do
+      echo "====== Cleanup Antrea-Agent Installation on the $host_name VM ======"
+      echo "Revert to snapshot VMAgent"
+      USERNAME=${LINUX_HOSTS_TO_USERNAME[${host_name}]}
+      IP_ADDRESS=${LINUX_HOSTS_TO_IP[${host_name}]}
+      govc snapshot.revert -k=true -vm=$host_name VMAgent
+      echo "Deleting stale antrea-agent files from $host_name"
+      ssh -o StrictHostKeyChecking=no -i "${WORKDIR}/jenkins_id_rsa" -n ${USERNAME}@${IP_ADDRESS} "rm -rf /tmp/antrea-ci"
+      # Temporary solution(till we update the snapshot)
+      ssh -o StrictHostKeyChecking=no -i "${WORKDIR}/jenkins_id_rsa" -n ${USERNAME}@${IP_ADDRESS} "sudo rm /etc/systemd/system/antrea-agent.service"
+      ssh -o StrictHostKeyChecking=no -i "${WORKDIR}/jenkins_id_rsa" -n ${USERNAME}@${IP_ADDRESS} "sudo systemctl stop antrea-agent"
+    done
+}
+
+function clean_vm_agent_on_windows {
+    for host_name in "${!WINDOWS_HOSTS_TO_IP[@]}"; do
+      echo "====== Cleanup Antrea-Agent Installation on the $host_name VM ======"
+      echo "Revert to snapshot VMAgent"
+      USERNAME=${WINDOWS_HOSTS_TO_USERNAME[${host_name}]}
+      IP_ADDRESS=${WINDOWS_HOSTS_TO_IP[${host_name}]}
+      govc snapshot.revert -k=true -vm=$host_name VMAgent
+      echo "Deleting stale antrea-agent files from $host_name"
+      ssh -o StrictHostKeyChecking=no -i "${WORKDIR}/jenkins_id_rsa" ${USERNAME}@${IP_ADDRESS} "rm -rf /tmp/antrea-ci"
+    done
+}
+
+function deliver_antrea_vm {
+    export_govc_env_var
+    clean_service_account_config
+    clean_vm_agent_on_linux
```

Comment:
Think about cleaning up the docker images before running test. Since you build both Windows and Linux binaries inside docker container on the same Node (master), and the size of Windows container image is huge, the disk size is always full after several times of tests without cleaup. And it might introduce unexpected build failures according to my previous experience.

Reply:
Done. Added in a clean antrea method.